packages/modules/pricing/src/services/pricing-module.ts

```diff
@@ -596,20 +598,48 @@ export default class PricingModuleService
 
     data?.forEach((price) => {
       const cleanRules = price.rules ? removeNullish(price.rules) : {}
-      const ruleEntries = Object.entries(cleanRules)
-      const rules = ruleEntries.map(([attribute, value]) => {
-        return {
-          attribute,
-          value,
-        }
-      })
+      const ruleOperators: PricingRuleOperatorValues[] =
+        Object.values(PricingRuleOperator)
+
+      const rules = Object.entries(cleanRules)
+        .map(([attribute, value]) => {
+          if (Array.isArray(value)) {
+            return value.map((customRule) => {
+              if (!ruleOperators.includes(customRule.operator)) {
+                throw new MedusaError(
+                  MedusaError.Types.INVALID_DATA,
+                  `operator should be one of ${ruleOperators.join(", ")}`
+                )
+              }
+
+              if (typeof customRule.value !== "number") {
+                throw new MedusaError(
+                  MedusaError.Types.INVALID_DATA,
+                  `value should be a number`
+                )
+              }
```

Comment:
comment: Wonder if this should live here, it significantly constrains the underlying rules engine of the pricing module. Would it make more sense for this to be at a higher level?